core/src/main/java/io/undertow/UndertowMessages.java

```diff
@@ -95,7 +95,7 @@
 //    IOException requestEntityWasTooLarge(SocketAddress address, long size);
 
     @Message(id = 20, value = "Connection terminated as request was larger than %s")
-    IOException requestEntityWasTooLarge(long size);
+    IllegalStateException requestEntityWasTooLarge(long size);
```

Comment:
This is not really correct, this exception can be thrown by any IO operation (not just form parsing), and it is not expected that read() operations will thrown IllegalStateException.